### PR TITLE
Hashed data - eduroam*,fedcloud*,feudal,firewall,flexlm*,freeipa

### PR DESCRIPTION
**Pull request by cuadradek**
- Services eduroam_block, eduroam_radiusm fedcloud_export, feudal,
firewall, flexlm_iptables and freeipa are now using
getHashedHierarchicalData.
- Services eduroam_block, feudal and freeipa were not manually tested.

### CODE REVIEW

File: gen/flexlm_iptables [hunk 1]
```diff
@@ -22,16 +22,16 @@ my $service_file_name = "$DIRECTORY/$::SERVICE_NAME";
 ####### output file ######################
 open SERVICE_FILE,">$service_file_name" or die "Cannot open $service_file_name: $! \n";
 
-my @resourcesData = $data->getChildElements;
-foreach my $rData (@resourcesData) {
-	my @membersData = $rData->getChildElements;
-	foreach my $mData (@membersData) {
-		my %memberAttributes = attributesToHash $mData->getAttributes;
-		$allowedIps{$_} = 1 foreach @{$memberAttributes{$A_U_IPS}};
+foreach my $resourceId ($data->getResourceIds()) {
```
Comment (zlamalp) [inline gen/flexlm_iptables:25]: If I understand this script correctly, it prints unique list of IP addresses taken from all users assigned to the service/facility. Then we can iterate over all members directly, using `getMemberIdsForFacility()`. Also can you add explanatory comment, what is printed to the output file. Code is a little hard to read and understand.

File: gen/feudal [hunk 2]
```diff
@@ -36,13 +36,9 @@ our $A_F_FEUDAL_ENTITLEMENT_SUFIX;      *A_F_FEUDAL_ENTITLEMENT_SUFIX =   \'urn:
...
-foreach my $resourceData ($data->getChildElements) {
-	my %resourceAttrs = attributesToHash $resourceData->getAttributes;
-
-	foreach my $groupData (($resourceData->getChildElements)[0]->getChildElements) {
-		my $groupMembersLogins = processGroupData $groupData;
+foreach my $resourceId ($data->getResourceIds()) {
+	foreach my $groupId ($data->getGroupIdsForResource(resource => $resourceId)) {
+		processGroupData $groupId, $resourceId;
 	}
 }
```
Comment (zlamalp) [inline gen/feudal:44]: Please add `->canonical` to JSON print, so its sorted and comparable between script runs.

File: gen/freeipa [hunk 3]
```diff
@@ -29,18 +29,18 @@ our $A_TITLE_AFTER;              *A_TITLE_AFTER =                 \'urn:perun:us
...
-foreach my $resourceData ($data->getChildElements) {
-	foreach my $groupData (($resourceData->getChildElements)[0]->getChildElements) {
-		processGroupData $groupData;
-		processGroupName $groupData, $groupNames;
+foreach my $resourceId ($data->getResourceIds()) {
+	foreach my $groupId ($data->getGroupIdsForResource(resource => $resourceId)) {
+		processGroupData $groupId, $resourceId;
+		processGroupName $groupId, $groupNames, $resourceId;
 	}
 }
```
Comment (zlamalp) [inline gen/freeipa:46]: Please add `->canonical` to JSON print.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: gen/flexlm_iptables (hunk 1 above)
Comment (cuadradek) [reply gen/flexlm_iptables]: Done

File: gen/feudal (hunk 2 above)
Comment (cuadradek) [reply gen/feudal:44]: Added

File: gen/freeipa (hunk 3 above)
Comment (cuadradek) [reply gen/freeipa:46]: Added

**Review by zlamalp — APPROVED** [review]

**Pull request merged**